topology/intel/dmic/dmic-nhlt.c

```diff
@@ -0,0 +1,1827 @@
+
+	char nhlt_string_buffer[MAX_NHLT_BUFFER_SIZE];
+	char *nhlt_p = &nhlt_string_buffer[0];
+	int i;
+
+	for (i = 0; i < size; i++) {
+		snprintf(nhlt_p, 12, "0x%08x,", *src);
+		nhlt_p += 11;
+		src++;
+	}
+
+	/* remove the last comma... */
+	nhlt_p--;
+	*nhlt_p = '\0';
+
+	printf("%s\n", nhlt_string_buffer);
+
+	tplg_pp_debug("");
+}
+
+static void dmic_print_internal(struct intel_nhlt_params *nhlt)
+{
+	int i, j, line, lines, remind;
+
+	tplg_pp_debug("printing dmic nhlt internal data");
+
+	/* top level struct */
+	tplg_pp_debug("gateway attributes: %u", nhlt->dmic_blob.gateway_attributes);
+
+	tplg_pp_debug("ts_group: %u %u %u %u", nhlt->dmic_blob.ts_group[0], nhlt->dmic_blob.ts_group[1],
+		      nhlt->dmic_blob.ts_group[2], nhlt->dmic_blob.ts_group[3]);
+
+	tplg_pp_debug("clock_on_delay: %u", nhlt->dmic_blob.clock_on_delay);
+
+	tplg_pp_debug("channel_ctrl_mask: %u", nhlt->dmic_blob.channel_ctrl_mask);
+
+	tplg_pp_debug("chan_ctrl_cfg: %u %u", nhlt->dmic_blob.chan_ctrl_cfg[0],
+		      nhlt->dmic_blob.chan_ctrl_cfg[1]);
+
+	tplg_pp_debug("channel_pdm_mask: %u", nhlt->dmic_blob.channel_pdm_mask);
+
+	/* first pdm struct */
+	tplg_pp_debug("pdm_ctrl_cfg 0");
+	tplg_pp_debug("cic_control: %u", nhlt->dmic_blob_pdm[0].cic_control);
+	tplg_pp_debug("cic_config: %u", nhlt->dmic_blob_pdm[0].cic_config);
+	tplg_pp_debug("mic_control: %u", nhlt->dmic_blob_pdm[0].mic_control);
+	tplg_pp_debug("pdmsm: %u", nhlt->dmic_blob_pdm[0].pdmsm);
+	tplg_pp_debug("reuse_fir_from_pdm: %u", nhlt->dmic_blob_pdm[0].reuse_fir_from_pdm);
+
+	/* first pdm struct, first fir */
+	tplg_pp_debug("fir_config 0");
+	tplg_pp_debug("fir_control: %u", nhlt->dmic_blob_fir[0][0].fir_control);
+	tplg_pp_debug("fir_config: %u", nhlt->dmic_blob_fir[0][0].fir_config);
+	tplg_pp_debug("dc_offset_left: %u", nhlt->dmic_blob_fir[0][0].dc_offset_left);
+	tplg_pp_debug("dc_offset_right: %u", nhlt->dmic_blob_fir[0][0].dc_offset_right);
+	tplg_pp_debug("out_gain_left: %u", nhlt->dmic_blob_fir[0][0].out_gain_left);
+	tplg_pp_debug("out_gain_right: %u", nhlt->dmic_blob_fir[0][0].out_gain_right);
+
+	/* first pdm struct, second fir */
+	tplg_pp_debug("fir_config 1");
+	tplg_pp_debug("fir_control: %u", nhlt->dmic_blob_fir[0][1].fir_control);
+	tplg_pp_debug("fir_config: %u", nhlt->dmic_blob_fir[0][1].fir_config);
+	tplg_pp_debug("dc_offset_left: %u", nhlt->dmic_blob_fir[0][1].dc_offset_left);
+	tplg_pp_debug("dc_offset_right: %u", nhlt->dmic_blob_fir[0][1].dc_offset_right);
+	tplg_pp_debug("out_gain_left: %u", nhlt->dmic_blob_fir[0][1].out_gain_left);
+	tplg_pp_debug("out_gain_right: %u", nhlt->dmic_blob_fir[0][1].out_gain_right);
+
+	/* first pdm struct, fir coeffs */
+	for (j = 0; j < DMIC_HW_CONTROLLERS; j++) {
+		tplg_pp_debug("fir_coeffs a length %u:", nhlt->dmic_fir_array.fir_len[0]);
+		lines = nhlt->dmic_fir_array.fir_len[0] / 8;
+		remind = nhlt->dmic_fir_array.fir_len[0] % 8;
+		for (i = 0; i < lines; i++) {
+			line = i * 8;
+			tplg_pp_debug("%d %u %u %u %u %u %u %u %u", i,
```

Comment:
These are signed integers so %d or hex would be better.

Reply:
ok